benchmarks/hexapdf_line_wrap/benchmark.rb

```diff
@@ -0,0 +1,51 @@
+require 'harness'
+
+# Before we activate Bundler, make sure gems are installed.
+Dir.chdir(__dir__) do
+  chruby_stanza = ""
+  if ENV['RUBY_ROOT']
+    ruby_name = ENV['RUBY_ROOT'].split("/")[-1]
+    chruby_stanza = "chruby && chruby #{ruby_name} && "
```

Comment:
This is an awkward workaround for some internal tooling, alas. The chruby call is needed because chruby isn't loaded when the script is, which loses track of which Ruby is supposed to be current (some env vars not propagated.)

So: you're right with normal external chruby. But just as we're hardcoding a nonstandard path for chruby, we're also working around nonstandard behaviour in the chruby install. It's annoying, and specific to Shopify developer laptops.

Reply:
It would probably make more sense to make the harnesses (run_benchmarks.rb here, yjit-metrics) look for a Gemfile and install automatically. Haven't done it yet though.

Reply:
Shouldn't the harness or ruby script already be run under `dev` or after `chruby`?
A script not propagating `PATH` seems pretty likely to go wrong for many things, including for any possible subprocess created by the benchmark if there was any.

Reply:
The problem is that it's looking in the wrong places for gems and gets the wrong version of Ruby. I'm not going to be able to share our internal modifications on this one. It has a number of odd features.

For instance, did you notice that "chruby && chruby #{ruby_name}"? That's because on the *first* call to chruby it doesn't pass the argument correctly, it just demand-loads chruby and prints the list of rubies. It needs a *second* call to chruby before you can get it to actually switch to the new Ruby version. It shouldn't do that that either, but it does, at least in this case.

I'm not going to be able to fully justify this to you without sharing our code, and the codebase in question isn't public.

Reply:
> Though that sounds like it would hit problems with native extensions.

Actually it wouldn't hit those issues, because the gems would be installed under the Ruby prefix and so never shared with other Rubies incorrectly. (The only issue is for CRuby dev, but that's a problem independently: https://bugs.ruby-lang.org/issues/18239#note-14)